packages/neon_framework/packages/neon_http_client/lib/src/interceptors/authentication_interceptor.dart

```diff
@@ -0,0 +1,72 @@
+import 'package:http/http.dart';
+import 'package:meta/meta.dart';
+import 'package:neon_http_client/src/interceptors/http_interceptor.dart';
+import 'package:nextcloud/nextcloud.dart';
+
+@internal
+final class AuthenticationInterceptor implements HttpInterceptor {
+  AuthenticationInterceptor({
+    required this.baseURL,
+  });
+
+  final Uri baseURL;
+  var _blocked = false;
+
+  bool _matchesBaseURL(Uri uri) => uri.toString().startsWith(baseURL.toString());
+
+  @override
+  bool shouldInterceptRequest(BaseRequest request) {
+    if (!_matchesBaseURL(request.url)) {
+      return false;
+    }
+
+    final authorization = request.headers['authorization'];
+    return authorization != null && authorization.isNotEmpty && _blocked;
+  }
+
+  @override
+  Never interceptRequest({required BaseRequest request}) {
+    assert(
+      shouldInterceptRequest(request),
+      'Request should not be intercepted.',
+    );
+
+    throw DynamiteStatusCodeException(Response('', 401));
```

Comment:
maybe the response message should mention that the interceptor blocked the request.

Reply:
Not really possible since the only parameter is the response and putting this message in the body field is wrong.